### PR TITLE
Fix incorrect maximum length of webadmin and webclient form fields

### PR DESCRIPTION
**Pull request by chris-allan**
Fixes incorrect maximum length of webadmin and webclient form fields. Having incorrect lengths causes scenarios where certain strings are truncated on save or action when used with these forms.

### CODE REVIEW

File: omeroweb/webadmin/forms.py [hunk 1]
```diff
@@ -67,7 +66,6 @@ def __init__(self, *args, **kwargs):
         super(ForgottonPasswordForm, self).__init__(*args, **kwargs)
         self.fields["server"] = ServerModelChoiceField(Server, empty_label=None)
         f = forms.CharField(
-            max_length=50,
```
Comment (sbesson) [inline omeroweb/webadmin/forms.py:70]: As this is a `username` should the max length limit be updated to 255 to match the database limit?

**Review by sbesson — COMMENTED** [review]
One question apart, all the updated length limitations in OMERO.web now match the size limits in the database with the current schema, namely:

- `Experimenter.omeName`, `Experimenter.firstname`, `Experimenter.middlename`, `Experimenter.lastname`, `Experimenter.email`, `Experimenter.institution` are columns with a VARCHAR(255) data type
- all container names e.g. `Project.name`, `Dataset.name` are columns with a TEXT data type
- `LogicalChannel.name` is a column with a VARCHAR(255) data type
- `LogicalChannel.fluor` is a column with a TEXT data type
- `Dichroic.lotnumber`, `Dichroic.serialNumber`, `Dichroic.manufacturer`, `Dichroic.model` are columns with a VARCHAR(255) data type
- `Microscope.lotnumber`, `Microscope.serialNumber`, `Microscope.manufacturer`, `Microscope.model` are columns with a VARCHAR(255) data type
- `Objective.lotnumber`, `Objective.serialNumber`, `Objective.manufacturer`, `Objective.model` are columns with a VARCHAR(255) data type
- `Filter.lotnumber`, `Filter.serialNumber`, `Filter.manufacturer`, `Filter.model`, Filter.filterwheel` are columns with a VARCHAR(255) data type
- `Detector.lotnumber`, `Detector.serialNumber`, `Detector.manufacturer`, `Detector.model` are columns with a VARCHAR(255) data type
- `Lightsource.lotnumber`, `Lightsource.serialNumber`, `Lightsource.manufacturer`, `Lightsource.model` are columns with a VARCHAR(255) data type


Finally `Password.hash` is a column with a VARCHAR(255) data type but that only applies to experiments which password is managed in the database. For all external authentication mechanisms like LDAP, this restriction does not necessary apply so dropping completely the `max_length` imposed in the OMERO.web form is the correct approach.

File: omeroweb/webadmin/forms.py (hunk 1 above)
Comment (chris-allan) [reply omeroweb/webadmin/forms.py:70]: Yes, fixed in 3acab1ee69c61fa8b2844d18134d38181017c975.

**Comment by chris-allan** [comment]
> One question apart

Did you have a question beyond the query about the username field in `ForgottonPasswordForm`?

> Finally Password.hash is a column with a VARCHAR(255) data type but that only applies to experiments which password is managed in the database.

Sort of. Its length is irrelevant regardless of the password source as the hash is of fixed length regardless of the size of the input data being hashed.

**Review by sbesson — APPROVED** [review]
> Did you have a question beyond the query about the username field in ForgottonPasswordForm?

No, the question on the ForgottenPassword form username length was my only outstanding question.

> Sort of. Its length is irrelevant regardless of the password source as the hash is of fixed length regardless of the size of the input data being hashed.

True i.e. Yet-another-reason to lift all length restrictions imposed on the user password.

From my side, all looks good from a code review.

**Review by will-moore — APPROVED** [review]
I wasn't sure if this needs any functional testing (or what's the easiest way to show the original errors)? If you exceed max_length when filling in a form, the form is invalid, so that should fail at validation (if we do it) rather than populating the DB with truncated values?

But the code changes look good. 👍

**Comment by chris-allan** [comment]
> I wasn't sure if this needs any functional testing (or what's the easiest way to show the original errors)?

The easiest way to see it is to create something like this:

```
omero obj new Project name=$(python -c 'print("a" * 250 + "b", end="")')
```

You will now not be able to modify the name of this object in due to form validation. However, since the field actually has 251 characters on the client, if you copy and paste the name back into the name field or into another name field it will be silently truncated. There are lots of other weird examples if you modify text in the middle of the long string, etc.

The worst example is the password field where you cannot actually see what's typed in, people are used to visually truncated password fields, and silent truncation happens underneath on paste.

**Comment by will-moore** [comment]
Thanks. Tested locally with this branch and could update e.g. Project name to a string longer than 250 characters.
Editing the same project when I reverted to master branch wasn't possible with the full name. Looks good.

**Comment by will-moore** [comment]
This seems to be included in merged PRs today at https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/478/console but I'm not seeing it at https://merge-ci.openmicroscopy.org/web/webclient/ 
E.g. Edit Project `<input>` still has `maxlength="250"`. Something up with the build...?

**Comment by will-moore** [comment]
This is being deployed on merge-ci now and is working fine.

**Pull request merged**